src/java.base/share/classes/sun/security/ssl/SSLKeyExchange.java

```diff
@@ -27,28 +27,43 @@
 
 import java.io.IOException;
 import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.HashMap;
+import java.util.List;
+import java.util.ListIterator;
 import java.util.Map;
 import sun.security.ssl.SupportedGroupsExtension.SupportedGroups;
 import sun.security.ssl.X509Authentication.X509Possession;
 
 final class SSLKeyExchange implements SSLKeyAgreementGenerator,
         SSLHandshakeBinding {
-    private final SSLAuthentication authentication;
+    private final List<SSLAuthentication> authentication;
     private final SSLKeyAgreement keyAgreement;
 
-    SSLKeyExchange(X509Authentication authentication,
+    SSLKeyExchange(List<X509Authentication> authentication,
             SSLKeyAgreement keyAgreement) {
-        this.authentication = authentication;
+        if (authentication != null) {
+            this.authentication = new ArrayList<>();
+            this.authentication.addAll(authentication);
```

Comment:
I may use an immutable list, for example List.copyOf(authentication).

Reply:
That's probably a good idea.  It doesn't look like we're making changes to the List outside of the constructor.